src/redux/books/booksSlice.js

```diff
@@ -0,0 +1,23 @@
+const initialState = {
+  books: [],
+};
+
+function booksReducer(state = initialState, action) {
+  switch (action.type) {
+    case 'ADD_BOOK':
+      return { ...state, books: [...state.books, action.payload] };
+    case 'REMOVE_BOOK':
+      return { ...state, books: state.books.filter((_, index) => index !== action.payload.index) };
+    case 'UPDATE_PROGRESS':
+      return {
+        ...state,
+        books: state.books.map((book, i) => (i === action.payload.index
+          ? { ...book, progress: Math.min(book.progress + 5, 100) }
+          : book)),
+      };
+    default:
+      return state;
+  }
+}
+
+export default booksReducer;
```

Comment:
- [ ] As the previous reviewer mentioned, the reducers should be set up with `createSlice` from redux toolkit. You can see [this tutorial](https://www.softkraft.co/how-to-setup-slices-with-redux-toolkit/) for create slice and how to export the reducers.